src/caching/IPFS/PinataIPFSClient.ts

```diff
@@ -0,0 +1,92 @@
+import { buildIPFSClient, retrieveValueFromIPFS, storeValueInIPFS } from "../../utils/IPFSUtils";
+import { CachingMechanismInterface } from "../../interfaces";
+import { Struct } from "superstruct";
+import winston from "winston";
+import PinataClient from "@pinata/sdk";
+import { jsonReviverWithBigNumbers, jsonReplacerWithBigNumbers, formattedLog } from "../../utils";
+
+/**
+ * A client for interacting with the IPFS. This is a wrapper around the IPFS API.
+ * This client also is a part of the caching mechanism interface.
+ * @see https://docs.ipfs.io/reference/http/api/
+ */
+export class PinataIPFSClient implements CachingMechanismInterface {
+  /**
+   * The IPFS client instance
+   */
+  private client: PinataClient;
+
+  /**
+   * Public IPFS gateway URL to retrieve the content from
+   */
+  private publicGatewayURL: string;
+
+  /**
+   * An optional logger for logging messages
+   */
+  private logger?: winston.Logger;
+
+  public constructor(projectId: string, projectSecret: string, publicGatewayURL: string, logger?: winston.Logger) {
+    this.client = buildIPFSClient(projectId, projectSecret);
+    this.publicGatewayURL = publicGatewayURL;
+    this.logger = logger;
+  }
+
+  /**
+   * Calls to a public IPFS gateway to retrieve a value.
+   * @param key The key to retrieve.
+   * @param _structValidator An optional struct validator to validate the retrieved value. If the value does not match the struct, null is returned.
+   * @returns The value if it exists, otherwise null.
+   */
+  async get<ObjectType>(key?: string, _structValidator?: Struct<unknown, unknown>): Promise<ObjectType | null> {
+    if (!key) {
+      return null;
+    }
+    formattedLog(this.logger, {
```

Comment:
I don't like this pattern that much, do we have to use `formattedLog` every time we log?

Reply:
My understanding is that this is an initial pass of how we can move toward a cohesive formatting tool. I'll definitely workshop it, but I think it's a good medium if you'd like to roll with it for now